examples/LoRaWAN/LoRaWAN_TS_Packages/LoRaWAN_TS009.h

```diff
@@ -0,0 +1,236 @@
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_TX_FRAMES_CTRL): {
+      switch(dataDown[1]) {
+        case(0):
+          // no change
+          // isConfirmed = isConfirmed;
+          break;
+        case(1):
+          isConfirmed = false;
+          break;
+        case(2):
+          isConfirmed = true;
+          break;
+      }
+      RADIOLIB_DEBUG_PRINTLN("Confirmed: %d", isConfirmed);
+
+      reply = false;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_ECHO_PAYLOAD): {
+      lenUp = lenDown;
+      for (int i = 1; i < lenDown; i++) {
+        dataUp[i] = dataDown[i] + 1;
+      }
+      fPort = RADIOLIB_LORAWAN_FPORT_TS009;
+      RADIOLIB_DEBUG_PRINTLN("Echoing payload");
+
+      reply = true;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_RX_APP_CNT): {
+      lenUp = 3;
+      uint16_t aFcntDown16 = (uint16_t)node.getAFCntDown();
+      dataUp[1] = aFcntDown16 & 0xFF;
+      dataUp[2] = aFcntDown16 >> 8;
+      fPort = RADIOLIB_LORAWAN_FPORT_TS009;
+      RADIOLIB_DEBUG_PRINTLN("aFCntDown16: %d", aFcntDown16);
+
+      reply = true;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_RX_APP_CNT_RESET): {
+      RADIOLIB_DEBUG_PRINTLN("Resetting Application Frame count");
+      node.resetFCntDown();
+
+      reply = false;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_LINK_CHECK): {
+      lenUp = 0;
+      fPort = RADIOLIB_LORAWAN_FPORT_MAC_COMMAND;
+      node.sendMacCommandReq(RADIOLIB_LORAWAN_MAC_LINK_CHECK);
+      RADIOLIB_DEBUG_PRINTLN("Requesting LinkCheck");
+
+      reply = true;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_DEVICE_TIME): {
+      lenUp = 0;
+      fPort = RADIOLIB_LORAWAN_FPORT_MAC_COMMAND;
+      node.sendMacCommandReq(RADIOLIB_LORAWAN_MAC_DEVICE_TIME);
+      RADIOLIB_DEBUG_PRINTLN("Requesting DeviceTime");
+
+      reply = true;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_PING_SLOT_INFO): {
+      lenUp = 0;
+      RADIOLIB_DEBUG_PRINTLN("Requesting PingSlotInfo not implemented");
+      // send PingSlotInfo MAC command which is not implemented
+      reply = false;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_TX_CW): {
+      // not implemented
+      uint16_t timeout = (dataDown[2] << 8) | dataDown[1];
+      uint32_t freqRaw = (dataDown[5] << 16) | (dataDown[4] << 8) | (dataDown[3]);
+      float freq = (float)freqRaw/10000.0;
+      uint8_t txPower = dataDown[6];
+      RADIOLIB_DEBUG_PRINTLN("Continuous wave: %7.3f MHz, %d dBm, %d s", freq, txPower, timeout);
+      radio.setFrequency(freq);
+      radio.setOutputPower(txPower);
+      radio.transmitDirect();
+      delay(timeout * 1000);
+      radio.standby();
+
+      reply = false;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_DUT_FPORT224_DISABLE): {
+      RADIOLIB_DEBUG_PRINTLN("Disabling FPort 224");
+      node.removePackage(RADIOLIB_LORAWAN_PACKAGE_TS009);
+
+      reply = false;
+    } break;
+
+    case(RADIOLIB_LORAWAN_TS009_DUT_VERSIONS): {
+      lenUp = 13;
+      // firmware version
+      dataUp[1] = 7;
```

Comment:
I think this is suposed to be the RadioLib version as an example - if so, I would add a comment explaning that. Also you can use the `RADIOLIB_VERSION_MAJOR` etc. macros here.